spec/gp_update_metadata_source_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::GpUpdateMetadataSourceAction do
+  it 'updates any block in a given .po file with the values from the given sources' do
+    Dir.mktmpdir do |dir|
+      # 1: Create a dummy .po file to use as input.
+      output_path = File.join(dir, 'output.po')
+      dummy_text = <<~PO
+        msgctxt "v1.0-whats-new"
+        msgid "this value should change"
+        msgstr ""
+        msgctxt "release_note_0122"
+        msgid "previous version notes required to have current one added"
+        msgstr ""
+        msgctxt "key1"
+        msgid "this value should change"
+        msgstr ""
+        msgctxt "key2"
+        msgid "this value should change"
+        msgstr ""
+      PO
+      File.write(output_path, dummy_text)
+
+      # 2: Create source files with value to insert in the .po
+      release_notes_path = File.join(dir, 'release_notes.txt')
+      File.write(release_notes_path, "- release notes\n- more release notes")
+      whats_new_path = File.join(dir, 'whats_new.txt')
+      File.write(whats_new_path, "- something new\n- something else new")
+      file_1_path = File.join(dir, '1.txt')
+      File.write(file_1_path, 'value 1')
+      file_2_path = File.join(dir, '2.txt')
+      File.write(file_2_path, 'value 2')
+      file_3_path = File.join(dir, '3.txt')
+      File.write(file_3_path, 'value 3')
+
+      described_class.run(
+        po_file_path: output_path,
+        release_version: '1.23',
+        source_files: {
+          release_note: release_notes_path,
+          whats_new: whats_new_path,
+          key1: file_1_path,
+          key2: file_2_path,
+          key3: file_3_path # This is not in the input .po and won't be added
```

Comment:
I wonder what we should do here to not forget to fix that.

I mean on one hand I'd want this test to go red, because our goal is ultimately to ensure all keys get included (see also the case of `release_notes_short` for Android which gets skipped because of this), and the refactoring to fix that will probably re-generate the `.po` file from scratch instead of trying to modify it in place.

On the other hand, you wrote those tests to kind of document the current, existing (buggy) behavior…

One option that could be nice would be to duplicate that test with the expected result (the one that do include `key3` as a new key added in the `.po` file), but [use `xit` instead of `it` to make that test disabled for now](https://www.rubyguides.com/2018/07/rspec-tutorial/#How_to_Temporarily_Disable_a_Test). It would then be easy to turn the test for current behavior from `it` to `xit` and the new one from `xit` to `it` to switch between testing old and new behaviors.

Reply:
Today I just discovered some other alternatives to mark a test as "pending" in RSpec: https://relishapp.com/rspec/rspec-core/v/2-0/docs/pending/pending-examples

This makes the test be shown in yellow in the terminal (instead of green for passed or red for failed).

I noticed it while I was working on something else and had a method in my helper do a `raise NoMethodError, "Not Implemented Yet"`, and the test that I wrote first before getting to that implementation (TDD rulz!) shown this when I ran `rspec`:

```
Pending: (Failures listed here are expected and do not affect your suite's status)

  1) FastlaneCore::Helper::Ios::L10nHelper#read_strings_file_as_hash 
     # Not yet implemented
     # ./spec/ios_l10n_helper_spec.rb:52
```

I thought it was a nice discovery and feature, and figured that you could take advantage of that here :)

Reply:
I like the pending idea. I'll add a dedicated test for the desired behavior and mark it as pending.